next-app/src/routes/password/change/+page.server.ts

```diff
@@ -0,0 +1,33 @@
+import { sf, type Rpc } from '$lib/server/sf'
+import { fetch_current_user } from '$lib/server/user'
+import { fail, redirect } from '@sveltejs/kit'
+import type { RequestEvent } from '../$types'
+
+export const actions = {
+  default: async ({ request }: RequestEvent) => {
+	const submissions = await request.formData()
+
+	const new_password = submissions.get('new_password')
+	const new_password_confirm = submissions.get('new_password_confirm')
+
+	if (! new_password) {
+		return fail(400, { failed: 'Password is required' })
+	}
+
+	if (new_password !== new_password_confirm) {
+		return fail(400, { failed: 'Passwords do not match' })
+	}
+
+	const cookie = request.headers.get('cookie') || ''
+
+	const { id } = await fetch_current_user(cookie)
+
+	await sf({
+		name: 'change_password',
+		args: [id, new_password],
```

Comment:
I get a compile error here that is fairly confusing, but ultimately it's because `new_password` isn't definitely a string and `args` needs to be either all strings or all objects. `submissions.get('new_password') as string` would be one way to go.
![image](https://user-images.githubusercontent.com/12587509/214811606-4ca3ff2d-761b-4282-96fa-1615bf12f3bd.png)

Reply:
yep, the `as string` is fine, corrected in [bf77994](https://github.com/sillsdev/web-languageforge/pull/1716/commits/bf77994b9fe002062e8e25cdbfac501aa3ddadc3)